Scripts/Sources/Scripts/main.swift

```diff
@@ -0,0 +1,162 @@
+else {
+    throw ScriptError.commandFailed("xcrun simctl list -j devices available failed to found the devices list")
+}
+
+let deviceList = try JSONDecoder().decode(SimulatorList.self, from: deviceListData)
+
+let availableDevices = deviceList.devices.flatMap { $0.value.map { $0.name } }
+
+for deviceName in devicesName {
+    if availableDevices.contains(deviceName) {
+        continue
+    }
+
+    print("     📲 \(deviceName) simulator is not available create it now")
+
+    guard shell(command: .xcrun, arguments: ["simctl", "create", deviceName, deviceName]).status == 0
+    else {
+        throw ScriptError.commandFailed("xcrun simctl create failed for the device \(deviceName)")
+    }
+}
+
+print("📺 Starting generating Marketing screenshots...")
+for deviceName in devicesName {
+    print("📱 Currently running on Simulator named: \(deviceName)")
+    print("     👷‍♀️ Generation of screenshots for \(deviceName) via test plan in progress")
+    print("     🐢 This usually takes some time...")
+
+    let iOSMarketingTestPlan = shell(command: .xcodebuild, arguments: [
+        "test",
+        "-scheme", "MemoArt (iOS)",
+        "-destination", "platform=iOS Simulator,name=\(deviceName)",
+        "-derivedDataPath", derivedDataPath,
+        "-testPlan", "Marketing",
+    ])
+
+    guard iOSMarketingTestPlan.status == 0 else {
+        print("Marketing UITests failed with errors")
+        print(iOSMarketingTestPlan.output ?? "Cannot print xcodebuild errors...")
+        continue
+    }
+    print("     ✅ Generation of screenshots for \(deviceName) via test plan done")
+
+    print("     👷‍♀️ Extraction and renaming of screenshots for \(deviceName) in progress")
+
+    let path = "\(derivedDataPath)/Logs/Test/LogStoreManifest.plist"
+
+    guard let manifestPlist = try? String(contentsOf: URL(fileURLWithPath: path), encoding: .utf8)
+    else {
+        print("Error, cannot read manifest Plist for this path: \(path)")
+        continue
+    }
+
+    guard let extractXCResultFileRegExp = try? NSRegularExpression(
+        pattern: "<key>fileName</key>.*?<string>(.*?)</string>",
+        options: NSRegularExpression.Options.dotMatchesLineSeparators
+    )
+    else {
+        print("Error, cannot build the regular expression to extract the XCResult file")
+        continue
+    }
+
+    let range = NSRange(location: 0, length: manifestPlist.utf16.count)
+    let xcresultFileNames: [String] = extractXCResultFileRegExp.matches(in: manifestPlist, options: [], range: range).flatMap { match in
+        (1..<match.numberOfRanges).map { rangeIndex in
+            let captureRange = match.range(at: rangeIndex)
+            let lowerIndex = manifestPlist.utf16.index(manifestPlist.startIndex, offsetBy: captureRange.lowerBound)
+            let upperIndex = manifestPlist.utf16.index(manifestPlist.startIndex, offsetBy: captureRange.upperBound)
+            return String(manifestPlist.utf16[lowerIndex..<upperIndex]) ?? "Error"
+        }
+    }
```

Comment:
Could be easily improved by using a XML parser library instead of using RegExp. Maybe not on this PR.

Reply:
Issue created, see #15